hazelcast-client-new/src/main/java/com/hazelcast/client/connection/nio/ClientReadHandler.java

```diff
@@ -52,41 +51,32 @@ public void run() {
     }
 
     @Override
-    public void handle() {
+    public void handle() throws Exception {
         lastHandle = Clock.currentTimeMillis();
         if (!connection.isAlive()) {
             if (logger.isFinestEnabled()) {
-                String message = "We are being asked to read, but connection is not live so we won't";
-                logger.finest(message);
+                logger.finest("We are being asked to read, but connection is not live so we won't");
             }
             return;
         }
-        try {
-            int readBytes = socketChannel.read(buffer);
+
+        int readBytes = socketChannel.read(buffer);
+        if (readBytes <= 0) {
             if (readBytes == -1) {
                 throw new EOFException("Remote socket closed!");
             }
-        } catch (IOException e) {
-            handleSocketException(e);
             return;
         }
-        try {
-            if (buffer.position() == 0) {
```

Comment:
Hi Sancar, they are already taken care of by using the readCount. So no need to check the readBuffer. The same approach is being used on the serverside.